model/role.go

```diff
@@ -484,12 +484,16 @@ func (r *Role) IsValidWithoutId() bool {
 
 	for _, permission := range r.Permissions {
 		permissionValidated := false
-		for _, p := range append(AllPermissions, DeprecatedPermissions...) {
-			if permission == p.Id {
-				permissionValidated = true
-				break
+		check := func(perms []*Permission, permission string) {
+			for _, p := range perms {
+				if permission == p.Id {
+					permissionValidated = true
+					break
+				}
 			}
 		}
+		check(AllPermissions, permission)
+		check(DeprecatedPermissions, permission)
```

Comment:
Maybe we could do the second check only if `permissionValidated` is `false`. If we make `check()` return a `bool` it might even simplify things further.